### ISSUE TITLE
Local build fails

### ISSUE THREAD

**Issue by danieljuschus** [mamba-org/micromamba-docker#251]
After cloning the repo (`git clone https://github.com/mamba-org/micromamba-docker.git`) and cding into it, I tried to build the image locally with `docker build .`. This failed with the following error:

```
> [stage-1  6/10] RUN /usr/local/bin/_dockerfile_initialize_user_accounts.sh &&     /usr/local/bin/_dockerfile_setup_root_prefix.sh:
#13 0.389 /bin/sh: 1: /usr/local/bin/_dockerfile_initialize_user_accounts.sh: not found
```

I'm a beginner with Docker, so I apologise if I'm doing something wrong.

**Comment by maresb** [comment]
> After cloning the repo (`git clone https://github.com/mamba-org/micromamba-docker.git`) and cding into it, I tried to build the image locally with `docker build .`. This failed with the following error:

That's odd. I just tested from a fresh clone and it worked for me. Could you try removing the cloned directory and trying again fresh?

**Comment by danieljuschus** [comment]
Just tried it again, no luck: `git clone https://github.com/mamba-org/micromamba-docker.git`, `cd micromamba-docker`, `docker build .` => `/bin/sh: 1: /usr/local/bin/_dockerfile_initialize_user_accounts.sh: not found`

Using Windows 11, Powershell 7.3.1 and Docker 20.10.21.

**Comment by maresb** [comment]
Hmm, I wonder if this is Windows weirdness with executable permissions. Could you try modifying the command in question to

```Dockerfile
RUN chmod a+x /usr/local/bin/_dockerfile_initialize_user_accounts.sh && \
    /usr/local/bin/_dockerfile_initialize_user_accounts.sh && \
    /usr/local/bin/_dockerfile_setup_root_prefix.sh
```

BTW, doing Docker stuff from Windows is a nightmare. Is booting into Linux an option for you? That way you can use the full resources of your system's kernel without the virtualization layer.

**Comment by wholtz** [comment]
My windows computer isn't accessible at the moment, so I won't be able to dig into this for a few days.

As @maresb mentioned, working in Linux is going to make working with docker easier. You might consider using the [Linux Subsystem for Windows (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install). It is a rather quick way to get access to a Linux system when running Windows.

**Comment by wholtz** [comment]
I have been able to replicate this issue. My first attempts at fixing it have not worked, but I will continue to look into this.

**Comment by wholtz** [comment]
@danieljuschus - I believe this has been fixed in #266.  The problem was line endings differences on Windows and Linux. With default settings, `git` will convert unix line endings into Windows line endings automatically. But docker doesn't automatically convert them back to unix line endings when you copy a file into an image. So I have configured the git repo to always keep unix line endings on shell scripts (`*.sh`).

Please do the following to test the fix:
1. delete your local `micromamba-docker` git repo directory
2. re-clone the repo
3. build

Please report back and let us know it worked for you.

**Comment by danieljuschus** [comment]
Working now on Windows, thanks a lot!
